karyon3-core/src/main/java/com/netflix/karyon/KeyMatchers.java

```diff
@@ -0,0 +1,51 @@
+package com.netflix.karyon;
+
+import java.lang.annotation.Annotation;
+
+import com.google.inject.Key;
+import com.google.inject.matcher.AbstractMatcher;
+import com.google.inject.matcher.Matcher;
+
+/**
+ * Utility class for creating matchers for Key<?>
+ */
+public final class KeyMatchers {
```

Comment:
Won't this lead to use of statics in library code?  Could the KeyMatcher be injected somehow?